### ISSUE TITLE
Remote Query Execution with sumIf() function faling

### ISSUE THREAD

**Issue by sidanasparsh** [ClickHouse/ClickHouse#18210]
I'm using clickhouse 20.12.3.3 version. 
The execution of sumIf() function is failing on the distributed table
Query:
`SELECT sumIf(pos_sales,isNotNull (pos_sales)) , fin_seg_desc AS SBU　from stores_cost_position_dist　where ((week between 201945 AND 202043)) group by fin_seg_desc;
`

Fails with:
`Code: 42. DB::Exception: Received from localhost:9000. DB::Exception: Aggregate function sum requires single argument: while receiving packet from node-2:9000: While executing Remote. 
`

Data types:
`pos_sales: Nullable(Float64),
week: Int32,
fin_seg_desc: String
`
The same query is working on the local table.

**Comment by den-crane** [comment]
repro

```sql
create table t ( a Nullable(Float64)) Engine=Memory;

SELECT sumIf(a, isNotNull(a)) 
from remote('127.0.0.1,127.0.0.2', currentDatabase(), t);

DB::Exception: Aggregate function sum requires single argument

```


But `sumIf(a, isNotNull(a))`  === `sum(a)` ?

**Comment by azat** [comment]
>repro

@den-crane Thanks you for the reproducer and bisect!

>related: #16574 ?

Indeed, it has bee introduced in #16610

Fix is here - #18806

**Comment by den-crane** [comment]
fix was never backported to 20.8

**Comment by den-crane** [comment]
oops, it was backported in another pr.

```sql
20.8.12.2


CREATE TABLE t
(
    `a` Nullable(Float64)
)
ENGINE = Memory

SELECT sumIf(a, isNotNull(a))
FROM remote('127.0.0.1,127.0.0.2', currentDatabase(), t)

Ok.

0 rows in set. Elapsed: 0.002 sec.
```
